operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -697,18 +747,115 @@ private void addIncludesSearchParameters(String compartmentMemberType, Multivalu
     }
     
     private void addSearchParameterIfNotExcluded(String compartmentMemberType, String code, 
-        ResourceType.Value subResourceType, MultivaluedMap<String, String> searchParameters, 
-        List<String> allowedIncludes) {
+            ResourceType.Value subResourceType, MultivaluedMap<String, String> searchParameters, 
+            List<String> allowedIncludes, List<String> extraResources, ParametersMap supportedSearchParametersMap) {
         // Need to make sure the search parameter has not been excluded
         String parameterName = compartmentMemberType + ":" + code + ":" + subResourceType.value();
         String simplifiedParameterName = compartmentMemberType + ":" + code;
-        if (allowedIncludes == null || allowedIncludes.contains(parameterName)
-                || allowedIncludes.contains(simplifiedParameterName)) {
+        // Also search if parameter name is supported by looking at the list of search
+        // parameters
+        SearchParameter searchParam = supportedSearchParametersMap.lookupByCode(code);
+        if ((extraResources != null && extraResources.contains(subResourceType.value())) &&
+                (allowedIncludes == null || allowedIncludes.contains(parameterName)
+                || allowedIncludes.contains(simplifiedParameterName)) && 
+                searchParam != null) {
             searchParameters.add("_include", parameterName);
         } else {
             if (LOG.isLoggable(Level.FINE)) {
                 LOG.fine("Filtering out searchParameter because it is not supported by the server config: " + parameterName);
             }
         }
     }
+    
+    /**
+     * Read additional associated resources for the "extra" types (like location) 
+     * supported by this server configuration
+     *
+     * @param compartmentMemberType the type of resource currently drilling down on
+     * @param newEntries a list of entries of that compartment type that were found in a search
```

Comment:
the "newEntries" parameter is a little confusing to me and the description only helped a little.
also "compartment type" still throws me off.
```suggestion
     * @param newEntries a list of Bundle response entries that include compartmentMemberType instances found via search
```

Reply:
Changed the javadoc as suggested and renamed compartmentType.